src/PrestaShopBundle/Twig/TranslationsExtension.php

```diff
@@ -0,0 +1,220 @@
+<?php
+/**
+ * 2007-2016 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\Twig;
+
+class TranslationsExtension extends \Twig_Extension
+{
+    /**
+     * @var \Symfony\Component\Translation\TranslatorInterface
+     */
+    public $translator;
+
+    /**
+     * Returns a list of functions to add to the existing list.
+     *
+     * @return array An array of functions
+     */
+    public function getFunctions()
+    {
+        return array(
+            new \Twig_SimpleFunction('getTranslationsTree', array($this, 'getTranslationsTree')),
+        );
+    }
+
+    /**
+     * Returns a tree of translations key values.
+     *
+     * @param array $translationsTree
+     * @return string
+     */
+    public function getTranslationsTree(array $translationsTree)
+    {
+        $output = '';
+        end($translationsTree);
+        list($lastTranslationDomain) = each($translationTree);
+        reset($translationTree);
+
+        foreach ($translationsTree as $topLevelDomain => $tree) {
+            $output .= $this->concatenateSubtreeHeader($topLevelDomain, $tree);
+
+            if ($lastTranslationDomain !== $topLevelDomain) {
+                $output .= '<hr />';
+            }
+        }
+
+        return $output;
+    }
+
+    /**
+     * @param $tree
+     * @return string|\Symfony\Component\Translation\MessageCatalogue
+     */
+    public function makeSubtree($tree, $level = 3)
+    {
+        $output = '';
+        $messagesSubtree = $this->hasCamelizedDomain($tree);
+
+        if ($messagesSubtree) {
+            $camelizedDomain = $tree['__camelized_domain'];
+            unset($tree['__camelized_domain']);
+
+$inputTemplate = <<<FORM
```

Comment:
I really don't like `HEREDOC`s, is it possible to get this template another way?